pkg/tidydns/tidydns.go

```diff
@@ -84,15 +112,248 @@ const (
 	RecordTypeSSHFP RecordType = 8
 	RecordTypeTLSA  RecordType = 9
 	RecordTypeCAA   RecordType = 10
+
+	AuthGroupUser       AuthGroup = 2
+	AuthGroupSuperAdmin AuthGroup = 1
 )
 
+const errorTidyDNS = "error from tidyDNS server: %s"
+const headerContentType = "Content-Type"
+const mimeForm = "application/x-www-form-urlencoded"
+
 type tidyDNSClient struct {
 	client   *http.Client
 	username string
 	password string
 	baseURL  string
 }
 
+func (c *tidyDNSClient) CreateInternalUser(ctx context.Context, username string, password string, description string, changePasswordOnFirstLogin bool, authGroup AuthGroup, userAllow []UserAllowID) (UserID, error) {
+	var userAllowFormatted []string
+	if len(userAllow) > 0 {
+		userAllowFormatted = make([]string, 0, len(userAllowFormatted))
+		for _, id := range userAllow {
+			userAllowFormatted = append(userAllowFormatted, strconv.Itoa(int(id)))
+		}
+	} else {
+		userAllowFormatted = []string{""}
+	}
+
+	var changePasswordOnFirstLoginFormatted string
+	if changePasswordOnFirstLogin {
+		changePasswordOnFirstLoginFormatted = "1"
+	} else {
+		changePasswordOnFirstLoginFormatted = "0"
+	}
+
+	data := url.Values{
+		"username":                       {username},
+		"epassword":                      {password},
+		"epassword_verify":               {password},
+		"change_password_on_first_login": {changePasswordOnFirstLoginFormatted},
+		"description":                    {description},
+		"auth_group":                     {strconv.Itoa(int(authGroup))},
+		//"tmp_auth_group":                 {""},
+		"user_allow": userAllowFormatted,
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", fmt.Sprintf("%s/=/user/new", c.baseURL), strings.NewReader(data.Encode()))
+	if err != nil {
+		return 0, err
+	}
+	req.SetBasicAuth(c.username, c.password)
+	req.Header.Set(headerContentType, mimeForm)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return 0, err
+	}
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(res.Body)
```

Comment:
This embedded function could be separated out to a proper function, since it's used often

Reply:
Fixed in 7b5d42d726b9bf22133aae712230e3f6e6d8ccec